nifi-nar-bundles/nifi-extension-utils/nifi-event-transport/src/main/java/org/apache/nifi/event/transport/netty/codec/SocketByteArrayMessageDecoder.java

```diff
@@ -38,7 +48,37 @@ public class SocketByteArrayMessageDecoder extends MessageToMessageDecoder<byte[
     protected void decode(final ChannelHandlerContext channelHandlerContext, final byte[] bytes, final List<Object> decoded) {
         final InetSocketAddress remoteAddress = (InetSocketAddress) channelHandlerContext.channel().remoteAddress();
         final String address = remoteAddress.getHostString();
-        final ByteArrayMessage message = new ByteArrayMessage(bytes, address);
+
+        final SslSessionStatus sslSessionStatus = getSslSessionStatus(channelHandlerContext);
+        final ByteArrayMessage message = new ByteArrayMessage(bytes, address, sslSessionStatus);
+
         decoded.add(message);
     }
+
+    private SslSessionStatus getSslSessionStatus(final ChannelHandlerContext channelHandlerContext) {
+        final Iterator<Map.Entry<String, ChannelHandler>> iterator = channelHandlerContext.channel().pipeline().iterator();
+        while (iterator.hasNext()) {
+            final ChannelHandler channelHandler = iterator.next().getValue();
+            if (channelHandler instanceof SslHandler) {
+                return createSslSessionStatusFromSslHandler((SslHandler) channelHandler);
+            }
+        }
+        return null;
```

Comment:
This could be refactored to use a single return statement. ChannelPipeline also implements `Iterable`, so it could be streamlined.
```suggestion
        SslHandler sslHandler = null;
        for (final Map.Entry<String, ChannelHandler> entry : channelHandlerContext.channel().pipeline()) {
            final ChannelHandler channelHandler = entry.getValue();
            if (channelHandler instanceof SslHandler) {
                sslHandler = (SslHandler) channelHandler;
                break;
            }
        }
        return sslHandler == null ? createSslSessionStatusFromSslHandler(sslHandler);
```

Reply:
Thank you! I applied your suggested change.